src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/reports/ReportsService.java

```diff
@@ -72,12 +73,18 @@ public ReportsService(
         String container,
         Classification classification
     ) {
+        List<Predicate<ZipFileSummaryResponse>> predicates = new ArrayList<>();
+        predicates.add(summary -> isEmpty(container) || summary.container.equalsIgnoreCase(container));
+        if (classification != null) {
+            predicates.add(summary -> classification.name().equalsIgnoreCase(summary.classification));
```

Comment:
predicate.and(summary -> classification.name().equalsIgnoreCase(summary.classification));
if you define predicate instead of array list.